library/std/src/sys/pal/unix/fs.rs

```diff
@@ -398,11 +401,12 @@ impl core::hash::Hash for FileType {
     }
 }
 
-#[derive(Debug)]
 pub struct DirBuilder {
     mode: mode_t,
 }
 
+struct Mode(mode_t);
```

Comment:
is there a reason for not using `Mode` as a field and continuing to derive the impls?